server/src/main/java/org/opensearch/indices/IndicesRequestCacheDiskTierPolicy.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.indices;
+
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.search.query.QuerySearchResult;
+
+import java.io.IOException;
+
+/**
+ * This policy takes in an array containing an instance of all policies we want to apply to the IRC disk tier.
+ * It short circuits these policies' checks together, in the provided order, to get one overall check.
+ */
+public class IndicesRequestCacheDiskTierPolicy implements CacheTierPolicy<QuerySearchResult> {
+    private final CacheTierPolicy<QuerySearchResult>[] policies;
+    private final int numPolicies;
+    private final boolean allowedByDefault;
```

Comment:
Why do we need this? If there are no policies mentioned ie list is empty, it should be allowedByDefault?